docs/source/index.rst

```diff
@@ -25,6 +25,105 @@ Contents:
 Upgrade Notes
 =============
 
+Upgrading to 1.12.0
+-------------------
+
+What Changed
+~~~~~~~~~~~~
+
+The botocore event system was changed to emit events based on the service id
+rather than the endpoint prefix or service name.
+
+Why Was The Change Was Made
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This was done to handle several issues that were becoming increasingly
+problematic:
+
+* Services changing their endpoint prefix would cause some registered events to
+  no longer fire (but not all).
+* New services that launch using an endpoint that another service is using
+  won't be able to be uniquely selected. There are a number of cases of this
+  already.
+* Services whose client name and endpoint prefix differed would require two
+  different strings if you want to register against all events.
+
+How Do I Know If I'm Impacted
```

Comment:
What about cases where the service id changed?  Not technically impacted but it might be good to document which event names have changed with the switchover to service ids so customers can start migrating their code and not rely on the aliasing (which will eventually go away).

Reply:
I do mention that briefly later on, but I can make it more explicit.